### PR TITLE
Set `no image` image instead of empty background

### LINKED ISSUE #2770: Display `No image` image instead of empty background if an image failed to load or for some other reason

### PR DESCRIPTION
**Pull request by Marc-Spector**
Closes #2770

### CODE REVIEW

**Comment by Marc-Spector** [comment]
image
image
image

**Comment by Marc-Spector** [comment]
It seems to work well. No memory leaks. :)

File: src/main/java/com/faforever/client/fx/ImageViewHelper.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          JavaFxUtil.removeListener(imageView.imageProperty(), this);
+        }
+      }
+    };
+
+    public void setDefaultImage(ImageView imageView, Image defaultImage, boolean onlyOnce) {
+      this.imageView = imageView;
+      this.defaultImage = defaultImage;
+      this.onlyOnce = onlyOnce;
+      JavaFxUtil.addListener(imageView.imageProperty(), onlyOnce ? imageListener : new WeakInvalidationListener(imageListener));
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/fx/ImageViewHelper.java]: I don't quite understand the only once parameter. If the listener is always  removed when completed why do we need to specify a weak listener?

Comment (Sheikah45) [reply src/main/java/com/faforever/client/fx/ImageViewHelper.java]: Also wouldn't we want to addAndTrigger all these listeners?

Comment (Sheikah45) [reply src/main/java/com/faforever/client/fx/ImageViewHelper.java]: And might be nice if possible if the imageViews started with the default image maybe.

Comment (Marc-Spector) [reply src/main/java/com/faforever/client/fx/ImageViewHelper.java]: I did a small refactoring of the class. Please review the code again.

Comment (Marc-Spector) [reply src/main/java/com/faforever/client/fx/ImageViewHelper.java]: > Also wouldn't we want to addAndTrigger all these listeners?

No need. It is extra.

Comment (Marc-Spector) [reply src/main/java/com/faforever/client/fx/ImageViewHelper.java]: > And might be nice if possible if the imageViews started with the default image maybe.

Done for some image views.

**Comment by Sheikah45** [comment]
Ah yeah that makes more sense now. 

The only thing I wonder now is if the image placeholder class should be made into a bean and have the UI service injected since it seems to be called the same way in a bunch of places.

**Comment by Marc-Spector** [comment]
Ok. This is the good suggestion, the code will become much cleaner.

**Review by Sheikah45 — APPROVED** [review]

**Pull request merged**